### PR TITLE
Use opensearch-ruby gem instead of elasticsearch one.

### LINKED ISSUE #3029: `rake environment elasticsearch:import:all` fails
## Steps to Reproduce

Follow [Development Setup](https://github.com/rubygems/rubygems.org/blob/master/CONTRIBUTING.md#development-setup) and run `bundle exec rake environment elasticsearch:import:all DIR=app/models FORCE=y` accordingly.

## Expected Behavior

```
[IMPORT] Loading models from: app/models
[IMPORT] Processing model: Rubygem...
[IMPORT] Done
```

## Current Behavior

```
[IMPORT] Loading models from: app/models
[IMPORT] Processing model: Rubygem...
rake aborted!
ArgumentError: wrong number of arguments (given 1, expected 0)
/Users/sato11/src/github.com/rubygems/rubygems.org/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.2.3/lib/active_record/relation/batches.rb:128:in `find_in_batches'
/Users/sato11/src/github.com/rubygems/rubygems.org/vendor/bundle/ruby/3.1.0/gems/activerecord-7.0.2.3/lib/active_record/querying.rb:22:in `find_in_batches'
/Users/sato11/src/github.com/rubygems/rubygems.org/vendor/bundle/ruby/3.1.0/gems/elasticsearch-model-7.1.1/lib/elasticsearch/model/proxy.rb:121:in `method_missing'
...
```

## Possible Solution

Apply `bundle update elasticsearch-rails` in local to let it pass. See #3028

## Additional Context

I was about to submit PR when I found it had already been done at #3028. Thank you @simi 
Submitting an issue is perhaps redundant but I bet it's better than nothing in case someone like me comes across 🙂

### PR DESCRIPTION
**Pull request by simi**
- update tracer setup in development
- use opensearch exceptions

fixes #3029

### CODE REVIEW

File: app/models/gem_download.rb [hunk 1]
```diff
@@ -93,7 +93,7 @@ def update_gem_downloads(updates_by_gem)
 
       # update ES index of rubygems
       Rubygem.__elasticsearch__.client.bulk body: bulk_update_query
-    rescue Faraday::ConnectionFailed, Elasticsearch::Transport::Transport::Error => e
+    rescue Faraday::ConnectionFailed, OpenSearch::Transport::Transport::Error => e
```
Comment (simi) [inline app/models/gem_download.rb:96]: :information_source: this seems untested, tests were green without this change.

**Comment by simi** [comment]
@sato11 you may be interested in this one as well. Feel free to test and ping back if everything works for elasticsearch import task.

**Comment by sato11** [comment]
As far as I'm concerned it looks ok 👍 Thanks for the heads up!

File: Gemfile.lock [hunk 2]
```diff
@@ -275,6 +275,14 @@ GEM
     nokogiri (1.13.4)
       mini_portile2 (~> 2.8.0)
       racc (~> 1.4)
+    opensearch-api (1.0.0)
+      multi_json
+    opensearch-ruby (1.0.0)
+      opensearch-api (= 1.0.0)
+      opensearch-transport (= 1.0.0)
```
Comment (sonalkr132) [inline Gemfile.lock:282]: We will have both opensearch-api, opensearch-transport and elasticsearch-api,elasticsearch-transport. Have you looked into which one is really being used? How are elasticsearch-* modules requiring opensearch-*? has opensearch overridden same classes?

**Comment by sonalkr132** [comment]
Gemfile.lock looks confusing with redudant gems. I am not super confident how stable this would be. IMHO, opensearch migration should be all or nothing. 
If purpose it just to setup elasticsearch db, one can use following. We can add our own rake task and update doc if needed.
```
Rubygem.__elasticsearch__.create_index! force: true
Rubygem.__elasticsearch__.import
```

**Comment by simi** [comment]
@sonalkr132 we're using following:

1. elasticsearch-rails (rails setup, rake tasks, ...)
2. elasticsearch-model (active record/model integration, depends on other elasticsearch gems, but client can be replaced as we do initializer)
3. elasticsearch-dsl (query builder)

We use initializer to assign Elasticsearch client to model.

This PR adds `opensearch-ruby` gem and assigns `OpenSearch` client to model. `elasticsearch-model` depends on other elastic gems, but in this PR we fetch explicitly `opensearch` gem. That way looking at `Gemfile` it should be clear:

### before

```ruby
gem "elasticsearch", "~> 7.10.1"
gem "elasticsearch-api", "~> 7.10.1"
gem "elasticsearch-transport", "~> 7.10.1"
gem "elasticsearch-model", "~> 7.0"
gem "elasticsearch-rails", "~> 7.0"
gem "elasticsearch-dsl", "~> 0.1.2"
```

### after

```ruby
gem "elasticsearch-model", "~> 7.2.1"
gem "elasticsearch-rails", "~> 7.2.1"
gem "elasticsearch-dsl", "~> 0.1.2"
gem "opensearch-ruby", "~> 1.0.0"
```

You can still find `elasticsearch-api`, `elasticsearch-transport` and `elasticsearch` gems in `Gemfile.lock` just because of those are dependencies of `elasticsearch-model`. But those gems are not used. OpenSearch gem doesn't rely on any `elasticsearch` gem and doesn't monkeypatch anything.

There is no guarantee elasticsearch-model will be compatible with newer versions of opensearch gem. We should migrate to real Elasticsearch service or go all in with OpenSearch and find out how to replace the missing elasticsearch-model functionality on our own (currently there is no opensearch-model gem).

**Comment by simi** [comment]
:information_source: updated after SearchKick migration. Once approved, we can test on staging with OpenSearch 1.3 upgrade.

**Review by jenshenny — APPROVED** [review]
Tested it out and looks good to me! Thanks @simi

**Pull request merged**